### ISSUE TITLE
very booring problem: after a while opening the same changing file, the content is non updated

### ISSUE THREAD

**Issue by luigif2000** [antonycourtney/tad#178]
Hi dear, and thank so much for Your amazing program, many compliment.
I meet the following problem
1) my script upgrade the same file with new data
2) i open the file and see the content, ok
3) after some time i reopen the file and i see new content, ok
4) the file changes time and length
5) after some time i reopen the file BUT i see the previous content file (point 3), 
6) the file continuing to change time and length
7) after that....if i open the file i continue to see the old content file (point 3)
..... and so on....for many times.

It seems a cached related problem, but not...other program as modern csv reader show the updated  content normally.......also vi and other editor

This is a very dangerous problem: you think the data remain unchanged and instead the data is changed!!!!
This happened to me many times.......and made me error thinking the data was not changed, when is really changed.

Please could You kindly help to solve this problem

thanks so much in advance .

luigi

**Comment by micka190** [comment]
Having the same problem on a 200+ row csv (some of the cells have paragraphs of text in them, so it's quite big). 
In my case, I have to close Tad and re-open it (simply re-opening the file doesn't work).

It'd be nice to have a way to disable whatever caching is being used (I wouldn't mind the 2-3 second load time whenever I made a change with an external application).

**Comment by tristanhoy** [comment]
@luigif2000 @micka190 the solution is to downgrade to [0.9.0](https://github.com/antonycourtney/tad/releases/tag/v0.9.0) which does not have this "feature"

![image](https://user-images.githubusercontent.com/11459446/210283269-faa83dec-c218-426c-b8af-e1ca240000ee.png)

It should really do a file timestamp check before loading from cache to make sure it's not stale.

**Comment by luigif2000** [comment]
Dear thanks a lot for Your kindly reply.
Sorry but i sinceresly don't understand: is it a problem on my side? Your side? is not a problem? 
is not possible, and dangerous, that i open a file, the file changes, i close the file, if i open again the file i see the old file!?
or not?

I'm confused, and downgrade is not so nice for me (loosing the latest features)

please let me kindly know better

thanks a lot in advance

luigi

**Comment by tristanhoy** [comment]
@luigif2000 The problem is with Tad. And I'm on Mac OS, so I'm not sure if other platforms are affected.

On Mac OS, when you "close" a file, it's not actually closing it - it's simply hiding it from view and the file stays in memory.

Even if you double-click on the file in Finder, it does not re-load the file - it just re-opens the version that's stuck in memory.

The only way to re-load a file and see changes is to completely quit Tad first (close ALL windows).

**Comment by tristanhoy** [comment]
Although I've just learned that downgrading to 0.9.0 on Mac means you can't paste values into filters, which is an even worse issue :(

**Comment by luigif2000** [comment]
Dear, thanks so much for your kindly reply.
On Linux also closing tad all windows doesn't solve the problem: i have to
rename the file....sigh.

I hope a solution soon ;-)

Bye

Luigi

Il Ven 6 Gen 2023, 22:37 tristanhoy ***@***.***> ha scritto:

**Comment by antonycourtney** [comment]
Hi Folks,
Just a quick note to let you know that I've seen the traffic on this issue, and agree this is a bug that should be fixed.
No promises on timing, but will do my best.
Sorry for the pain in the interim and thanks for your patience.

**Comment by luigif2000** [comment]
Thanks thanks so much for your awesome program *

Best regards

Luigi

Il Sab 7 Gen 2023, 18:53 Antony Courtney ***@***.***> ha
scritto:

**Comment by antonycourtney** [comment]
Hi @luigif2000 @tristanhoy @micka190 and others,
I know it's been a long wait on this issue, but I'm pleased to report that I just published Tad 0.12.0, which has basic support for this feature request. If you open a file that is already open in Tad, Tad will check the file modification time against what it was at time of import.  If the file modification is newer (indicating the file was updated), it will re-import the file.
You can download Tad 0.12.0 from the [Tad Releases page](https://github.com/antonycourtney/tad/releases/tag/v0.12.0).
Please give it a try and let me know if this solves the problem for you, or if you encounter any issues.
Thanks for the feedback!  Closing this issue for now.

**Comment by luigif2000** [comment]
Thanks thanks so much.....i see the email now.....asap i will check!

Best regards and wish you the best

Luigi

Il Lun 6 Mar 2023, 19:22 Antony Courtney ***@***.***> ha
scritto:
